scalableminds/webknossos
# Refactored several admin views into React functional components

React classifies `class` components as a [Legacy API](https://react.dev/reference/react/Component) and libraries like `antd` only offer some features through hooks, e.g. `const {modal } = App.useApp(); modal.confirm(...)`.

Hence, I started to refactor several admin views into React functional components to reduce our technical debt. I intentionally left out anything from the dashboard and main data viewer for the moment. Let's start with the less critical admin views first.

For the most part I only tried to convert the components 1:1. In some cases I did some light refactoring; especially the components including forms, since the antd API changes a little anyways.

In the process there should be a higher type coverage and some antd component styling fixed for static modal methods (`modal.confirm` etc).

I also slightly simplified the signature of `Utils.compareBy` and `Utils.localeCompareBy`. Both had a weird typehint argument that is not required by Typescript and likely a remains of Flow.

Note to reviewers on the most common changes:
- Removed a whole lot of `this`, `this.props` and `this.state`
- Replaced state from a class property with `useState` hook
- Replaced React class component life cycle methods with hook, typically `useEffect`or `useFetch`
- The content of `render` methods remain unchanged and are only un-indented. Except for the marked files I did not make any changes to the "DOM" structure.
- Replace class methods with functions

### URL of deployed dev instance (used for testing):
- https://reactfunctionalcomponents.webknossos.xyz/

### Steps to test:
- Navigate to all changes views and make sure they render correctly.
- Do some monkey testing for all the admin views. Create new stuff, delete, and edit it. 

### TODOs:
- [x] Fix remaining type errors
- [x] Fix remaining linting errors

### Issues:
- Contributes to #6861 

------
(Please delete unneeded items, merge only when none are left open)
